broker-core/src/main/java/io/zeebe/broker/workflow/processor/flownode/PropagateTerminationHandler.java

```diff
@@ -20,27 +20,28 @@
 import io.zeebe.broker.workflow.model.element.ExecutableFlowNode;
 import io.zeebe.broker.workflow.processor.BpmnStepContext;
 import io.zeebe.broker.workflow.processor.BpmnStepHandler;
-import io.zeebe.broker.workflow.processor.EventOutput;
 import io.zeebe.broker.workflow.state.ElementInstance;
 import io.zeebe.protocol.intent.WorkflowInstanceIntent;
 
 public class PropagateTerminationHandler implements BpmnStepHandler<ExecutableFlowNode> {
   @Override
   public void handle(BpmnStepContext<ExecutableFlowNode> context) {
-    final EventOutput output = context.getOutput();
     final ElementInstance flowScopeInstance = context.getFlowScopeInstance();
+    final int activeExecutionPaths = flowScopeInstance.getNumberOfActiveExecutionPaths();
 
-    if (flowScopeInstance.getNumberOfActiveElementInstances() == 0) {
-      if (flowScopeInstance.isInterrupted()) {
-        context
-            .getCatchEventOutput()
-            .triggerInterruptedElement(flowScopeInstance, context.getOutput().getStreamWriter());
-      }
+    if (activeExecutionPaths > 0) {
```

Comment:
Won't this be true if not all children elements have finished terminating? This is the sum of active tokens + active child elements, so if we have a parallel gateway in a sub-process, when the first branch terminates it will execute this handler, and this will be true no? Then we'd already trigger the boundary event before all children elements are terminated, I think.

Reply:
No, it works ;) When the first element is terminated then `activeExecutionPaths > 0` but there are no deferred tokens for this element. So it will not trigger the boundary event.